src/spac/data_utils.py

```diff
@@ -497,45 +496,45 @@ def downsample_cells(data, annotation, n_samples=None,
 
     Returns
     -------
-    data : pd.DataFrame
+    output_data : pd.DataFrame
         The downsampled data frame.
 
     Examples
     --------
     >>> df = pd.DataFrame({
-    ...    'annotation1': ['a', 'a', 'b', 'b', 'c', 'c'],
-    ...    'annotation2': ['x', 'y', 'x', 'y', 'x', 'y'],
+    ...    'annotations1': ['a', 'a', 'b', 'b', 'c', 'c'],
+    ...    'annotations2': ['x', 'y', 'x', 'y', 'x', 'y'],
     ...    'value': [1, 2, 3, 4, 5, 6]
     ... })
     >>> print(
-            downsample_cells(df, ['annotation1', 'annotation2'], n_samples=2)
+            downsample_cells(df, ['annotations1', 'annotations2'], n_samples=2)
         )
     """
 
-    # Convert annotation to list if it's a string
-    if isinstance(annotation, str):
-        annotation = [annotation]
+    # Convert annotations to list if it's a string
+    if isinstance(annotations, str):
+        annotations = [annotations]
 
-    # Check if the columns to downsample on exists
-    for col in annotation:
-        if col not in data.columns:
+    # Check if the columns to downsample on exist
+    for col in annotations:
+        if col not in input_data.columns:

```

Comment:
@fangliu117 , can we use the util function to check for annotations?

Reply:
Yes, update it.

Reply:
Update: I tried to use util function, which requires the Anndata as input, while for most of functions of data_utils, we have inputs as dataframe. Since this function is for data preprocessing, and to keep the consistence, I leave these lines now, but if needed, I'd like do more modification.

Reply:
Thanks @fangliu117, yes your comments make sense. You don't need to update these lines.